ChangeLog.d/fix_build_error_for_mbedtls_deprecated_removed.txt

```diff
@@ -0,0 +1,3 @@
+Bugfix
+    * Fix build error due to missing prototype
+      warning when MBEDTLS_DEPRECATED_REMOVED is enabled
```

Comment:
What can we do to ensure this doesn't happen again? `MBEDTLS_DEPRECATED_REMOVED` is tested via `component_test_default_no_deprecated` in `tests/scripts/all.sh`, which is passing. Should we maybe use clang instead of gcc?

Reply:
I can also reproduce this with gcc 10 and `-Wmissing-prototypes` (I haven't gone back in versions to see when that first came in)

Reply:
... and with `gcc (Ubuntu 8.4.0-3ubuntu2) 8.4.0`

Reply:
... and `gcc (Ubuntu 7.5.0-6ubuntu2) 7.5.0` (2017)

Reply:
`gcc-7 $ make CC=gcc CFLAGS='-Wall -Werror -Wmissing-prototypes'`

Reply:
We should probably raise a separate issue to track this so we can merge & close this PR

Reply:
@gilles-peskine-arm `MBEDTLS_DEPRECATED_REMOVED` just checks that the build is consistent, it doesn't check for missing prototypes.

Following @daverodgman's suggestion, I've raised #6179, so we can get this PR in and then ensure it doesn't happen again